src/mapeo-project.js

```diff
@@ -124,4 +180,36 @@ export class MapeoProject {
   get field() {
     return this.#dataTypes.field
   }
+
+  /**
+   * @param {Partial<Omit<import('@mapeo/schema').ProjectValue, 'schemaName'>>} settings
+   * @returns {Promise<import('@mapeo/schema').Project>}
+   */
+  async $setProjectSettings(settings) {
+    const { project } = this.#dataTypes
+
+    const existing = await project.getByDocId(this.#projectId).catch(() => {
+      // project does not exist so return null
+      return null
+    })
+
+    if (existing) {
+      return project.update([existing.versionId, ...existing.forks], {
+        ...valueOf(existing),
+        ...settings,
+      })
+    }
+
+    return project[kCreateWithDocId](this.#projectId, {
+      ...settings,
+      schemaName: 'project',
+    })
+  }
+
+  /**
+   * @returns {Promise<Omit<import('@mapeo/schema').Project, 'schemaName'>>}
+   */
+  async $getProjectSettings() {
+    return this.#dataTypes.project.getByDocId(this.#projectId)
```

Comment:
Catch and return `{}`

Reply:
any objections to returning `null` here instead?

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/187/commits/a6a468f2dfc6c688975d9ebead97b8364b0cfbdf. went ahead and returned `null` instead since it makes more sense to me 😄

Reply:
hm starting to see why you suggested `{}`. doing the `null` checks after calling this may get annoying. still think i prefer it to `{}` and the code you'd have to write to account for it, but open to alternative reasoning

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/187/commits/fe56a1a52d29eb48742649adb418e9ead3a8bf06